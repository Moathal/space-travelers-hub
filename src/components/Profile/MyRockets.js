import React from 'react';
import { useSelector } from 'react-redux';
import MyRocket from './MyRocket';

function MyRockets() {
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div>
      <h2>My Rockets</h2>
      { reservedRockets.map((rocket) => (
        <MyRocket key={rocket.id} name={rocket.rocket_name} />
      ))}
    </div>
  );
}

export default MyRockets;
