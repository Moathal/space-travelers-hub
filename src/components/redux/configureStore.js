import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions';
import rocketsReducer from './rockets';

export default configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice,
  },
});
