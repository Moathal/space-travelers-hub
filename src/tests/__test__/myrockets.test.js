import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../components/redux/configureStore';
import MyRockets from '../../components/Profile/MyRockets';

it('renders MyRockets correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <MyRockets />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
