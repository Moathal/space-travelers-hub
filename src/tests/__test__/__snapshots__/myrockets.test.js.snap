// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MyRockets correctly 1`] = `
<div>
  <h2>
    My Rockets
  </h2>
</div>
`;
